Rubberduck.UnitTesting/UnitTesting/TestCodeGenerator.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NLog;
+using Rubberduck.Interaction;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Resources.UnitTesting;
+using Rubberduck.SettingsProvider;
+using Rubberduck.SmartIndenter;
+using Rubberduck.UnitTesting.Settings;
+using Rubberduck.VBEditor.SafeComWrappers;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.UnitTesting.UnitTesting
+{
+    public partial class TestCodeGenerator : ITestCodeGenerator
+    {
+        protected static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+        private readonly bool _isAccess;
+        private readonly IConfigProvider<UnitTestSettings> _settings;
+        private readonly RubberduckParserState _state;
+        private readonly IMessageBox _messageBox;
+        private readonly IIndenter _indenter;
+        private readonly IVBEInteraction _interaction;
+
+        public TestCodeGenerator(IVBE vbe, RubberduckParserState state, IMessageBox messageBox, IVBEInteraction interaction, IConfigProvider<UnitTestSettings> settings, IIndenter indenter)
+        {
+            _isAccess = "AccessApp".Equals(vbe?.HostApplication()?.GetType().Name);
+            _state = state;
+            _messageBox = messageBox;
+            _interaction = interaction;
+            _settings = settings;           
+            _indenter = indenter;
+        }
+
+        public void AddTestModuleToProject(IVBProject project)
+        {
+            AddTestModule(project, null);
+        }
+
+        public void AddTestModuleToProject(IVBProject project, Declaration stubSource)
+        {
+            AddTestModule(project, stubSource);
+        }
+
+        private void AddTestModule(IVBProject project, Declaration stubSource)
+        {
+            if (project == null || project.IsWrappingNullReference)
+            {
+                return;
+            }
+
+            var settings = _settings.Create();
+
+            if (settings.BindingMode == BindingMode.EarlyBinding)
+            {
+                // FIXME: Push the actual adding of TestModules into UnitTesting, which sidesteps VBEInteraction being inaccessble here
+                _interaction.EnsureProjectReferencesUnitTesting(project);
+            }
+
+            try
+            {
+                using (var components = project.VBComponents)
+                using (var component = components.Add(ComponentType.StandardModule))
+                using (var module = component.CodeModule)
+                {
+                    component.Name = GetNextTestModuleName(project);
+
+                    // Test modules always have appropriate options so remove any pre-generated code.
+                    if (module.CountOfLines > 0)
+                    {
+                        module.DeleteLines(1, module.CountOfLines);
+                    }
```

Comment:
That deletes all code in the module.  The component is newly added, so the only code it can possibly contain at that point are the auto-generated options from the VBE.